shapely/geos.py

```diff
@@ -93,9 +93,16 @@ def load_dll(libname, fallbacks=None, mode=DEFAULT_MODE):
     # Test to see if we have a delocated wheel with a GEOS dylib.
     geos_whl_dylib = os.path.abspath(os.path.join(os.path.dirname(
         __file__), '.dylibs/libgeos_c.1.dylib'))
+
     if os.path.exists(geos_whl_dylib):
-        _lgeos = CDLL(geos_whl_dylib)
-        LOG.debug("Found GEOS DLL: %r, using it.", _lgeos)
+        handle = CDLL(None)
+        if hasattr(handle, "initGEOS_r"):
+            LOG.debug("GEOS already loaded")
+            _lgeos = handle
```

Comment:
I did not realize this about CDLL, but called with `None` it gets all the currently loaded symbols in the process. We can discover whether libgeos has already been loaded, such as via libgdal in the case where fiona or rasterio is imported before shapely. If it has, we don't try to load the wheel's included dynamic library.